commands.js

```diff
@@ -1485,6 +1489,7 @@ exports.commands = {
 			let modchat = Tools.escapeHTML(room.modchat);
 			this.add("|raw|<div class=\"broadcast-red\"><b>Moderated chat was set to " + modchat + "!</b><br />Only users of rank " + modchat + " and higher can talk.</div>");
 		}
+		if (room.battle && !room.modchat && !user.can('modchat')) room.requestModchat(null);
```

Comment:
They have room permissions. This checks global modchat ability (as far as I can tell)